### PR TITLE
Extract encrypt/decrypt refactor from state PR

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Extracted from #64

Implements a generic way to encryption and decryption of structs.

## Code changes

- Organization id changed to optional.
- Added Encryptable & Decryptable trait to support encrypting and decrypting multiple data types.
- Update existing places to use the new traits.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)

### CODE REVIEW

File: crates/bitwarden/src/secrets_manager/secrets/secret_response.rs [hunk 1]
```diff
@@ -31,9 +32,21 @@ impl SecretResponse {
...
+        let value = response
+            .value
+            .ok_or(Error::MissingFields)?
+            .parse::<CipherString>()?
+            .decrypt(enc, &org_id)?;
+        let note = response
+            .note
+            .ok_or(Error::MissingFields)?
+            .parse::<CipherString>()?
+            .decrypt(enc, &org_id)?;
```
[the remark this answers is not in the document]
Comment (dani-garcia) [reply crates/bitwarden/src/secrets_manager/secrets/secret_response.rs:49]: I assumed that the note field would always be present, even if it's an empty string. The secret create request requires it at least, and the bws cli just sets it to an empty string when not specified: https://github.com/bitwarden/sdk/blob/master/crates/bitwarden-api-api/src/models/secret_create_request_model.rs.

If the field could be missing, we'd have to change this to:
```rust
        let note: Option<String> = response
            .note
            .map(|n| Ok(n.parse::<CipherString>()?.decrypt(enc, &org_id)?))
            .transpose()?;
```

[the remark this answers is not in the document]
Comment (Hinton) [reply crates/bitwarden/src/secrets_manager/secrets/secret_response.rs:49]: That seems to be an accurate assumption. Seems I miss-remembered how it worked. Not going to be the case for ciphers though and we should be vigilant.

**Review by MGibson1 — CHANGES_REQUESTED** [review]

File: crates/bitwarden/src/secrets_manager/secrets/secret_response.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hinton) [reply crates/bitwarden/src/secrets_manager/secrets/secret_response.rs:49]: I don't believe it's a regression though, but something for us to keep track of and try and iterate on.

File: crates/bitwarden/src/crypto.rs [hunk 2]
```diff
@@ -383,16 +383,16 @@ impl<T: Encryptable<Output>, Output, Id: Hash + Eq> Encryptable<HashMap<Id, Outp
     }
 }
 
-impl<T: Decryptable<Output>, Output, Id: Hash + Eq> Decryptable<HashMap<Id, Output>>
+impl<T: Decryptable<Output>, Output, Id: Hash + Eq + Copy> Decryptable<HashMap<Id, Output>>
```
Comment (MGibson1) [inline crates/bitwarden/src/crypto.rs:386]: 🤔 should this be `Clone`? `Copy` is good enough for the `Uuid` we use today, but is there a good reason it _has_ to be?

**Review by dani-garcia — APPROVED** [review]

File: crates/bitwarden/src/crypto.rs (hunk 2 above)
Comment (MGibson1) [reply crates/bitwarden/src/crypto.rs:386]: 🤦 got those backward

File: crates/bitwarden/src/secrets_manager/secrets/secret_response.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MGibson1) [reply crates/bitwarden/src/secrets_manager/secrets/secret_response.rs:49]: fair, it's not as far as I can tell

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**